build_modules/lib/src/errors.dart

```diff
@@ -37,3 +42,58 @@ class KernelSummaryException extends _WorkerException {
   KernelSummaryException(AssetId summaryId, String error)
       : super(summaryId, error);
 }
+
+/// An [Exception] that is thrown when there are some missing modules.
+class MissingModulesException implements Exception {
+  final String message;
+
+  @override
+  String toString() => message;
+
+  MissingModulesException._(this.message);
+
+  static Future<MissingModulesException> create(
+      Module module,
+      Set<AssetId> missingSources,
+      List<Module> transitiveModules,
+      AssetReader reader) async {
+    var buffer = new StringBuffer(
+        ' \n\nUnable to find modules for some sources, check the following '
```

Comment:
why do we need the preceding newlines here? That seems like it should be the responsibility of whatever is using this string.

Reply:
It looks nicer and these are directly logged in 3 places :man_shrugging: 

We could make this always happen for severe logs?